src/puya/awst_build/eb/app_account_state.py

```diff
@@ -115,31 +121,13 @@ def call(
     ) -> ExpressionBuilder:
         match args:
             case [item]:
-                app_local_get_ex = _build_app_local_get_ex(self.state_def, item, location)
+                field = _build_field(self.state_decl, item, location)
+                app_local_get_ex = StateGetEx(field=field, source_location=location)
                 return var_expression(app_local_get_ex)
             case _:
                 raise CodeError("Invalid/unhandled arguments", location)
 
 
-def _build_app_local_get_ex(
-    state_def: AppStateDefinition, item: ExpressionBuilder | Literal, location: SourceLocation
-) -> IntrinsicCall:
-    index_expr = _validated_index_expr(item)
-    app_local_get_ex = IntrinsicCall(
-        source_location=location,
-        op_code="app_local_get_ex",
-        stack_args=[
-            index_expr,
-            UInt64Constant(value=0, source_location=location),
-            BytesConstant(
-                value=state_def.key, encoding=state_def.key_encoding, source_location=location
-            ),
-        ],
-        wtype=wtypes.WTuple.from_types((state_def.storage_wtype, wtypes.bool_wtype)),
-    )
-    return app_local_get_ex
-
-
 def _validated_index_expr(index: ExpressionBuilder | Literal) -> Expression:
     # TODO: FIXME
```

Comment:
🤔

Reply:
Wasn't part of this PR but I've now simplified this anyway.